### PR TITLE
Create JSON output format

### LINKED ISSUE #96: JSON output
It would be really handy, to be able to do something with the returned output, in its current form while not impossible to awk and manipulate it. Would be ideal to have the output as a JSON optionally to be able to parse into other applications.

### PR DESCRIPTION
**Pull request by jejung**
A simple implementation of a JSON format output. Technically closes #96.

example

### CODE REVIEW

**Comment by sstalle** [comment]
Thank you for the PR, @jejung.

I have a couple of notes:

- `JsonResultPrinter` accumulates all line numbers and error messages in an array while the program is running. Though messages are usually short, a large number of them can still take up a considerable amount of memory. I would prefer messages to be printed immediately and not stored until the program finishes running.
- It would be nice to have checked file count and elapsed time in JSON output to keep consitency between different output formats.

File: src/Infrastructure/ContainerBuilder.php [hunk 1]
```diff
@@ -187,7 +194,13 @@ public function buildContainer(OutputInterface $output, $intSize)
             return new StandardPrettyPrinter();
         };
         $container['resultPrinter'] = function ($c) {
-            return new CLIResultPrinter($c['output'], $c['nodePrinter'], $c['nodeStatementsRemover']);
+            switch ($this->outputFormat) {
```
Comment (sstalle) [inline src/Infrastructure/ContainerBuilder.php]: Unfortunately `$this` cannot be used in a closure, because we still have to support PHP 5.3

**Comment by jejung** [comment]
Updated output format:

image

**Comment by sstalle** [comment]
Thanks for you work, @jejung . If you are going to use the version from master, please note that I renamed "messages" to "warnings" in JSON output to be consistent with the documentation.

**Pull request merged**